### PR TITLE
wb_robot_cleanup fix for python plugins

### PR DESCRIPTION
**Pull request by angel-ayala**
trying to solve #476 by adding a `stop()` method in python plugins

### CODE REVIEW

File: webots_ros2_driver/src/PythonPlugin.cpp [hunk 1]
```diff
@@ -37,6 +42,7 @@ from controller import Supervisor
 sys.path.insert(1, os.path.dirname(controller.__file__))
 from vehicle import Driver
 
+
```
Comment (ygoumaz) [inline webots_ros2_driver/src/PythonPlugin.cpp:45]:
```suggestion
```

**Review by ygoumaz — CHANGES_REQUESTED** [review]
Very nice! Thank you for the contribution. Could you please add an entry in both related changelogs? 

I will further investigate if we can terminate cleanly the supervisor node too.

**Comment by ygoumaz** [comment]
Have you tried to run some examples on Iron distribution (I am not sure if the following error is also on Humble), because I found a problem in examples that don't use Python plugins, namely `webots_ros2_tiago`.

If I start it:
```bash
ros2 launch webots_ros2_tiago robot_launch.py 
```

and then interrupt it with Ctrl+C, I get the following logs:
```logs
[webots_controller_Tiago_Lite-5] Warning: class_loader.ClassLoader: SEVERE WARNING!!! Attempting to unload library while objects created by this loader exist in the heap! You should delete your objects before attempting to unload the library or destroying the ClassLoader. The library will NOT be unloaded.
[webots_controller_Tiago_Lite-5]          at line 127 in ./src/class_loader.cpp
[webots_controller_Tiago_Lite-5] Warning: class_loader.ClassLoader: SEVERE WARNING!!! Attempting to unload library while objects created by this loader exist in the heap! You should delete your objects before attempting to unload the library or destroying the ClassLoader. The library will NOT be unloaded.
[webots_controller_Tiago_Lite-5]          at line 127 in ./src/class_loader.cpp
[webots_controller_Tiago_Lite-5] [ros2run]: Segmentation fault
[ERROR] [webots_controller_Tiago_Lite-5]: process has died [...]
```

It is not related to the changes of this PR but I wonder if we should not try to fix this issue at the same time. What do you think?

**Comment by angel-ayala** [comment]
I have not been able to test it in Iron, I will run it in Humble to see if it also happens.

**Comment by angel-ayala** [comment]
I was checking, that maybe the problem is related to the return added in `WebotsNode.step()`
https://github.com/cyberbotics/webots_ros2/blob/2ce3fb2fca040d72346ded64ff68fc05e13bb284/webots_ros2_driver/src/WebotsNode.cpp#L257
inside the condition when the signal is captured, I'm returning -1 without calling `wb_robot_step` [(code diff)](https://github.com/cyberbotics/webots_ros2/pull/781/files#diff-e4089e8690a190770b4f97a3d72cbfc73cca04b51934f118e633da76436b4971R259)
```python
if (gShutdownSignalReceived && !mWaitingForUrdfRobotToBeRemoved) {
  mRemoveUrdfRobotPublisher->publish(mRemoveUrdfRobotMessage);
  mWaitingForUrdfRobotToBeRemoved = true;
  return -1
}
```
once the SIGINT is received, should it call `wb_robot_step`? this is the only changed behavior that affects both Python and C++ plugins, what do you think @ygoumaz?

**Comment by ygoumaz** [comment]
This error also occurs in the master branch, so it doesn't seem to be related to the changes of this PR. We can already merge this to fix the termination of Python plugins and I will open an issue for C++ plugins on Iron.

**Comment by ygoumaz** [comment]
@angel-ayala, can you update the changelogs, so that we can merge this PR before releasing 2023.1.1?

**Comment by angel-ayala** [comment]
yeah sure, I was finishing some issues with my project's code. I will update this today.

**Review by ygoumaz — APPROVED** [review]
Thank you!

**Pull request merged**